### PR TITLE
Speeding up tools/update.sh by fetching repos without history or submodules

### PR DESCRIPTION
**Pull request by wmathurin**

### CODE REVIEW

File: tools/update.sh [hunk 1]
```diff
@@ -72,52 +62,35 @@ get_root_folder ()
...
-        echo "Cloning $git_repo_url into $repo_dir"
-        git clone ${git_repo_url} ${repo_dir}
-        cd ${repo_dir}
+        echo "Cloning $git_repo_url at $git_branch into $repo_dir"
+        git clone --branch $git_branch --single-branch --depth 1 $git_repo_url $repo_dir
     else
-        echo "Found repo at $repo_dir.  Fetching the latest"
-        cd ${repo_dir}
-        git fetch origin
+        echo "Found repo at $repo_dir.  Remove and run update.sh again."
```
Comment (wmathurin) [inline tools/update.sh:76]: Now will error out if the repo is already found => you need to start clean.

File: tools/update.sh [hunk 2]
```diff
@@ -72,52 +62,35 @@ get_root_folder ()
...
     local git_repo_url=$2
+    local git_branch=${OPT_BRANCH}
 
     if [ ! -d "$repo_dir" ]
     then
-        echo "Cloning $git_repo_url into $repo_dir"
-        git clone ${git_repo_url} ${repo_dir}
-        cd ${repo_dir}
+        echo "Cloning $git_repo_url at $git_branch into $repo_dir"
+        git clone --branch $git_branch --single-branch --depth 1 $git_repo_url $repo_dir
```
Comment (wmathurin) [inline tools/update.sh:74]: Cloning just the desired branch with depth 1.

File: tools/update.sh [hunk 3]
```diff
@@ -72,52 +62,35 @@ get_root_folder ()
...
 }
 
 ROOT_FOLDER=$(get_root_folder)
 ANDROID_SDK_REPO_PATH="https://github.com/forcedotcom/SalesforceMobileSDK-Android.git"
 ANDROID_SDK_FOLDER="SalesforceMobileSDK-Android"
-IOS_SDK_REPO_PATH="https://github.com/forcedotcom/SalesforceMobileSDK-iOS-Hybrid.git"
-IOS_SDK_FOLDER="SalesforceMobileSDK-iOS-Hybrid"
+IOS_HYBRID_SDK_REPO_PATH="https://github.com/forcedotcom/SalesforceMobileSDK-iOS-Hybrid.git"
+IOS_HYBRID_SDK_FOLDER="SalesforceMobileSDK-iOS-Hybrid"
+IOS_SDK_REPO_PATH="https://github.com/forcedotcom/SalesforceMobileSDK-iOS.git"
```
Comment (wmathurin) [inline tools/update.sh:86]: We are not getting submodules so we need to get SalesforceMobileSDK-iOS directly.
We assume here that the SalesforceMobileSDK-iOS-Hybrid in branch x uses SalesforceMobileSDK-iOS submodule at branch x also.

**Review by bhariharan — APPROVED** [review]

**Comment by wmathurin** [comment]
Now downloading 44 Mb instead of 620 Mb (on my local 29 seconds instead of 5 minutes).

**Pull request merged**